### PR TITLE
compile single css

### PR DESCRIPTION
**Pull request by AverageMarcus**
See #64 

Merged all CSS (inc 3rd party) into single minified file and removed all unused selectors.

Also inlined CSS and JS into HTML page. Total page size ~47KB

cc/ @spikeheap

### CODE REVIEW

**Comment by benfoxall** [comment]
bootstrap.css doesn't seem to be in the codebase (there's a `css/*.css` ignore rule in `.gitignore`)

**Comment by AverageMarcus** [comment]
Doh! I knew I needed to change that but totally forgot. Serves me right for coding at night. I'll get on it later

File: Gruntfile.js [hunk 1]
```diff
@@ -104,11 +103,22 @@ module.exports = function(grunt) {
...
+        files: {
+          '_site/index.html': '_site/index.html',
+        },
+      },
+    },
   });
 
-  grunt.registerTask('build', ['less','jekyll:build']);
-  grunt.registerTask('optimize', ['imagemin','cssmin','uglify','htmlmin']);
+  grunt.registerTask('build', ['less','jekyll:build','uncss','cssmin']);
```
Comment (benfoxall) [inline Gruntfile.js]: are `uncss` and `cssmin` not part of the optimise task?

Comment (AverageMarcus) [inline Gruntfile.js]: Sort of, but the min.css file is referenced by the html so it needs to be built.

Comment (benfoxall) [inline Gruntfile.js]: ah - cool.  That makes sense

**Comment by AverageMarcus** [comment]
added bootstrap and font-awesome

**Comment by spikeheap** [comment]
Hi @AverageMarcus, cheers for the PR. `grunt-html-smoosher` looks to be dead (no updated since Feb 2014), and is [broken](https://github.com/motherjones/grunt-html-smoosher/issues/23), though I see you fixed the uglify-js issue on your fork. I'm not sure it's wise to add-in a dead plugin: the future burden is likely much greater than the additional network overhead of grabbing Bootstrap from the CDN.

As we have a `less` step, the way I've solved this in the past is just to `include` Bootstrap  and build a master CSS file, and then let uncss work on the single compiled file, but there are many ways of skinning a :cat:.

Also, the Bootstrap files seem to be missing from this PR.

**Comment by AverageMarcus** [comment]
Is the fact that it's dead such an issue if it achieves the task needed? I guess there are alternatives that could be tried, simply chose this as I was familiar with it.

Using the less include is prob a better way to do it. Simplifies things considerably in the Gruntfile. I'll update that later tonight.

**Comment by spikeheap** [comment]
I've pulled Bootstrap/Fontawesome from the repo and used Bower to pull the deps down instead. While I was at it I've added the Bower and Gem dependencies to the NPM postinstall task.

**Pull request merged**